src/views/Add.vue

```diff
@@ -0,0 +1,176 @@
+      <template v-slot:title>
+        <div class="mx-auto text-center" style="font-size: 24px">
+          {{ pageTitle }}
+        </div>
+      </template>
+
+      <v-sheet width="300" class="mx-auto">
+        <v-select
+          v-model="selected"
+          :items="options"
+          label="Select resource type"
+        ></v-select>
+
+        <v-form @submit.prevent="handleSubmit" ref="form">
+          <Pearl v-if="selected === 'Pearl'" />
+          <PreciousMetal v-if="selected === 'PreciousMetal'" />
+          <Gemstone v-if="selected === 'Gemstone'" />
+          <LinkingPart v-if="selected === 'LinkingPart'" />
+
+
+          <v-card-text
+            v-if="sentData"
+            class="text-red-darken-4 text-center text-h6 pa-0"
+            style="height: 1rem"
+          >
+            {{
+              getText("Item has been updated", "Information has been submited")
+            }}
+          </v-card-text>
+          <div v-else style="height: 1rem"></div>
+
+          <div v-if="selected" class="d-flex flex-column">
+            <v-btn color="success" class="mt-4" block type="submit">
+              {{ getText("Edit", "Submit") }}
+            </v-btn>
+
+            <v-btn color="error" class="mt-4" block @click="resData">
+              Reset
+            </v-btn>
+
+            <v-btn color="warning" class="mt-4" block to="/resources">
+              Go Back
+            </v-btn>
+          </div>
+        </v-form>
+      </v-sheet>
+    </v-card>
+  </v-container>
+</template>
+
+<script>
+import { ref, computed, watch, inject } from "vue";
+import Pearl from "../components/AddResources/Pearl.vue";
+import PreciousMetal from "../components/AddResources/PreciousMetal.vue";
+import Gemstone from "../components/AddResources/Gemstone.vue";
+import LinkingPart from "../components/AddResources/LinkingPart.vue";
+import { useRoute } from "vue-router";
+import { useStore } from "vuex";
+
+export default {
+  components: {
+    Pearl,
+    PreciousMetal,
+    Gemstone,
+    LinkingPart
+  },
+  props: {
+    id: String,
+  },
+  setup(props) {
+    const store = useStore();
+    const route = useRoute();
+    const options = ref(["Pearl", "Gemstone", "PreciousMetal", "LinkingPart"]);
+    const pageTitle = ref(route.meta.title);
+    const getText = computed(() => (text1, text2) => {
+      return pageTitle.value === "Edit resource" ? text1 : text2;
+    });
+
+    const showSnackbar = inject("showSnackbar");
+    const selected = ref("");
+    const form = ref(null);
+    const sentData = ref(false);
+    const formData = computed(() => store.state.resources.resourceDetails);
+
+    if (props.id) {
+      const data = computed(() =>store.getters["resources/getResourceById"](props.id));
+      store.dispatch("resources/setResourceDetails", data.value);
+      const resourceDetail = computed(() => store.getters["resources/getResourceDetails"]);
+      selected.value = resourceDetail.value.clazz;
```

Comment:
Let's rename data to resourceDetails. We can also remove the `store.getters["resources/getResourceDetails"])` call as we already have the clazz field from the `store.getters["resources/getResourceById"](props.id)` object. 

The code would become:
```
const resourceDetails = computed(() =>store.getters["resources/getResourceById"](props.id));
store.dispatch("resources/setResourceDetails", resourceDetails.value);
selected.value = resourceDetails.value.clazz;
```